### PR TITLE
common/buffer.cc: use shift_round_up() when appropriate

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by rzarzynski — APPROVED** [review]

File: src/common/buffer.cc [hunk 1]
```diff
@@ -2276,8 +2277,10 @@ MEMPOOL_DEFINE_OBJECT_FACTORY(buffer::raw_static, buffer_raw_static,
 
 
 void ceph::buffer::list::page_aligned_appender::_refill(size_t len) {
-  const size_t alloc = \
-    std::max(static_cast<size_t>(min_alloc), static_cast<size_t>((len + CEPH_PAGE_SIZE - 1) & CEPH_PAGE_MASK));
+  const unsigned alloc =
+    std::max(min_alloc,
+	     shift_round_up(static_cast<unsigned>(len),
```
Comment (mattbenjamin) [inline src/common/buffer.cc:2282]: why are you preferring (pre-ansi) "unsigned" to an explicit stdint type like uint32_t?

Comment (tchaikov) [reply src/common/buffer.cc:2282]: good question, indeed. i didn't think about this. what i wanted is to be in line with the type of the parameter of `buffer::create_page_aligned()`.

actually, we already have 3 unsigned types here

- `unsigned`: in `buffer::create_page_aligned(unsigned len)`
- `size_t`: in `page_aligned_appender::_refill(size_t len)`
- `unsigned long`: `CEPH_PAGE_SHIFT`

i am not sure if we can improve the situation by introducing yet another (but probably better) unsigned type.

if i had the luxury to pick one and stick with it, i'd do so. but `buffer.h` is part of the public API. i'd prefer not changing them.

Comment (mattbenjamin) [reply src/common/buffer.cc:2282]: that make sense.  I think at some point, buffer should switch to using explicit, portable types, but it makes sense to keep consistent with the rest of buffer here

Comment (tchaikov) [reply src/common/buffer.cc:2282]: yes, *portable*! we had a bunch portable fixes in this area in the past to address the unportable mixture of `size_t`, `unsigned`  and `std::max()`.  as `std::max()` and friends is able to do template type deduce in C++17.

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-04-06_14:10:15-rados-wip-kefu-testing-2021-04-06-1610-distro-basic-gibba/

failures tracked by

- https://tracker.ceph.com/issues/50196
- https://tracker.ceph.com/issues/48995
- https://tracker.ceph.com/issues/49344

**Pull request merged**